runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -235,31 +251,33 @@ public int appendObjectState(Stream<Map.Entry> entryStream) {
         ImmutableMap<CheckpointEntry.CheckpointDictKey, String> mdkv =
                 ImmutableMap.copyOf(this.mdkv);
 
-        final Iterable<List<Map.Entry>> partitions =
-                Iterables.partition(entryStream::iterator, batchSize);
         int entryCount = 0;
-
-        for (List<Map.Entry> partition : partitions) {
-            MultiSMREntry smrEntries = new MultiSMREntry();
-            for (Map.Entry entry : partition) {
-                smrEntries.addTo(new SMREntry("put",
-                        new Object[]{keyMutator.apply(entry.getKey()),
-                                valueMutator.apply(entry.getValue())},
-                        serializer));
-                entryCount++;
+        int numBytesPerCheckpointEntry = 0;
+        ByteBuf b = Unpooled.buffer();
+        MultiSMREntry smrEntries = new MultiSMREntry();
+
+        for (Map.Entry entry : (Iterable<Map.Entry>) entryStream::iterator) {
```

Comment:
We removed the 50 entries limit here. We need to confirm with Maithem or someone else that this removal does not possibly affect anywhere else in Corfu.

Reply:
With regards to this, there are some tests that rely on this to verify an entry is of type continuation, those tests need to accordingly be changed, like for example: checkpointWriterInterleavedTest

Reply:
It did. The test"checkpointWriterInterleavedTest" was failing. After discussion with Maithem on the zoom call, I am going to maintain a local batch size counter, specially for the validity by the test.

Reply:
@annym Thanks for pointing it out. I am going to take care of it in the next update I am going to make to this pull request